.github/workflows/ci.yml

```diff
@@ -46,13 +47,15 @@ jobs:
 
         sudo python3 -m pip install --upgrade pip
         sudo pip3 install scikit-build cmake requests gitpython gcovr pyyaml
+        sudo curl https://sh.rustup.rs -sSf | sh -s -- -y
     - name: cmake
       env:
         CC: ${{ matrix.compiler.cc }}
         CXX: ${{ matrix.compiler.cxx }}
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # has to be included to access other secrets
         GITHUB_HUNTER_USERNAME: ${{ secrets.GITHUB_HUNTER_USERNAME }}
         GITHUB_HUNTER_TOKEN: ${{ secrets.GITHUB_HUNTER_TOKEN }}
+        CFLAGS: "-I /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include"
```

Comment:
Possible point to refactor:
needed for compiling of the Rust system libraries, without this environment variable we will not be able to find C headers (such as errno.h and so on) on MacOS. Linux builds are successful.

Reply:
Is there a way how to add that directory only when matrix.os is macOS?

Reply:
Found solution to add that directory only for macOS